orion/pkg/controllers/semester_controller_test.go

```diff
@@ -0,0 +1,211 @@
+	semesterService.mockGetBySemesterId = func(semesterId string) (domains.Semester, error) {
+		semester := createMockSemester("2020_fall", "Fall 2020")
+		return semester, nil
+	}
+	services.SemesterService = &semesterService
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/semesters/v1/semester/2020_fall", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+	var semester domains.Semester
+	if err := json.Unmarshal(recorder.Body.Bytes(), &semester); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+	assert.EqualValues(t, "2020_fall", semester.SemesterId)
+	assert.EqualValues(t, "Fall 2020", semester.Title)
+}
+
+func TestGetSemester_Failure(t *testing.T) {
+	semesterService.mockGetBySemesterId = func(semesterId string) (domains.Semester, error) {
+		return domains.Semester{}, errors.New("not found")
+	}
+	services.SemesterService = &semesterService
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/semesters/v1/semester/2020_winter", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusNotFound, recorder.Code)
+}
+
+//
+// Test Create
+//
+func TestCreateSemester_Success(t *testing.T) {
+	semesterService.mockCreate = func(semester domains.Semester) error {
+		return nil
+	}
+	services.SemesterService = &semesterService
+
+	// Create new HTTP request to endpoint
+	semester := createMockSemester("2020_fall", "Fall 2020")
+	marshal, _ := json.Marshal(semester)
+	body := bytes.NewBuffer(marshal)
+	recorder := sendHttpRequest(t, http.MethodPost, "/api/semesters/v1/create", body)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+}
+
+func TestCreateSemester_Failure(t *testing.T) {
+	// no mock needed
+	services.SemesterService = &semesterService
+
+	// Create new HTTP request to endpoint
+	semester := createMockSemester("2020_fall", "")
+	marshal, _ := json.Marshal(semester)
+	body := bytes.NewBuffer(marshal)
+	recorder := sendHttpRequest(t, http.MethodPost, "/api/semesters/v1/create", body)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusBadRequest, recorder.Code)
+}
+
+//
+// Test Update
+//
+func TestUpdateSemester_Success(t *testing.T) {
+	semesterService.mockUpdate = func(semesterId string, semester domains.Semester) error {
+		return nil // Succesful update
+	}
+	services.SemesterService = &semesterService
+
+	// Create new HTTP request to endpoint
+	semester := createMockSemester("2020_winter", "Winter 2020")
+	body := createBodyFromSemester(semester)
+	recorder := sendHttpRequest(t, http.MethodPost, "/api/semesters/v1/semester/2020_fall", body)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+}
+
+func TestUpdateSemester_Invalid(t *testing.T) {
+	// no mock needed
+	services.SemesterService = &semesterService
+
+	// Create new HTTP request to endpoint
+	semester := createMockSemester("2020_winter", "")
```

Comment:
nit: you can make a comment either here or in line 142 about why this endpoint is an invalid request (i.e. invalid semester name).